arangod/RestHandler/RestIndexHandler.cpp

```diff
@@ -61,12 +61,8 @@ LogicalCollection* RestIndexHandler::collection(
     std::string const& cName, std::shared_ptr<LogicalCollection>& coll) {
   if (!cName.empty()) {
     if (ServerState::instance()->isCoordinator()) {
-      try {
-        coll = ClusterInfo::instance()->getCollection(_vocbase.name(), cName);
-
-        return coll.get();
-      } catch (...) {
-      }
+      coll = ClusterInfo::instance()->getCollectionNT(_vocbase.name(), cName);
+      return coll.get();
```

Comment:
why is this method filling a shared_ptr passed by reference and at the same time returning a pointer to the object contained in the shared_ptr?
can the signature be changed so that this function just returns the shared_ptr?

Reply:
Probably the line below doesn't lend out the memory, but keeps it, so it mustn't be freed outside?
```
      return _vocbase.lookupCollection(cName).get();
    }
```

Reply:
Just make the method return a shared_ptr in all cases, instead of having it return a pointer and fill a shared_ptr by reference.